jsf/subsystem/src/main/java/org/jboss/as/jsf/deployment/JSFManagedBeanProcessor.java

```diff
@@ -111,6 +116,24 @@ public void deploy(final DeploymentPhaseContext phaseContext) throws DeploymentU
     }
 
     /**
+     * WFLY-6617 workaround
+     * According to  JSF 2.2 spec, it should be possible to inject beans using @EJB annotation into
+     * PhaseListeners.
+     */
+    private void handlePhaseListeners(CompositeIndex index, Set<String> managedBeanClasses) {
```

Comment:
This is wrong. You need to make phase listeners into components based on in they are enabled (i.e. listed in faces-config.xml AFAIK). Instead of looking for EJB injections you need to look at the parsed faces-config.xml data. This should also be in a different processor (or this one renamed to JSFComponentProcessor or something like that).

There are other types of injection other than @EJB that will not work with your fix (e.g. @DataSource).

Reply:
@stuartwdouglas Thanks, i ll fix it asap